_clients/ruby.md

```diff
@@ -0,0 +1,528 @@
+    { delete: { _index: 'students', _id: 200 } },
+    { delete: { _index: 'students', _id: 300 } }
+]
+client.bulk(body: actions, refresh: true)
+```
+
+You can perform different operations when using `bulk` as follows:
+
+```ruby
+actions = [
+    { index:  { _index: 'students', _id: 100, data: { first_name: 'Paulo', last_name: 'Santos', gpa: 3.29, grad_year: 2022 } } },
+    { index:  { _index: 'students', _id: 200, data: { first_name: 'Shirley', last_name: 'Rodriguez', gpa: 3.92, grad_year: 2020 } } },
+    { index:  { _index: 'students', _id: 300, data: { first_name: 'Akua', last_name: 'Mansa', gpa: 3.95, grad_year: 2022 } } },
+    { index:  { _index: 'students', _id: 400, data: { first_name: 'John', last_name: 'Stiles', gpa: 3.72, grad_year: 2019 } } },
+    { index:  { _index: 'students', _id: 500, data: { first_name: 'Li', last_name: 'Juan', gpa: 3.94, grad_year: 2022 } } },
+    { index:  { _index: 'students', _id: 600, data: { first_name: 'Richard', last_name: 'Roe', gpa: 3.04, grad_year: 2020 } } },
+    { update: { _index: 'students', _id: 100, data: { doc: { gpa: 3.73 } } } },
+    { delete: { _index: 'students', _id: 200  } }
+]
+client.bulk(body: actions, refresh: true)
+```
+
+In the above example, you pass the data and the header together and you denote the data with the `data:` key.
+
+## Searching for a document
+
+To search for a document, use the `search` method. The following example searches for a student whose first or last name is "James." It uses a `multi_match` query to search for two fields (`first_name` and `last_name`), and it is boosting the `last_name` field in relevance with a caret notation (`last_name^2`). 
+
+```ruby
+q = 'James'
+query = {
+  'size': 5,
+  'query': {
+    'multi_match': {
+      'query': q,
+      'fields': ['first_name', 'last_name^2']
+    }
+  }
+}
+
+response = client.search(
+  body: query,
+  index: 'students'
+)
+```
+
+If you omit the request body in the `search` method, your query becomes a `match_all` query and returns all documents in the index:
+
+```ruby
+client.search(index: 'students')
+```
+
+## Boolean query
+
+The Ruby client exposes full OpenSearch query capability. In addition to simple searches that use the match query, you can create a more complex Boolean query to search for students who graduated in 2022 and sort them by last name. In the example below, search is limited to 10 documents.
+
+```ruby
+query = {
+    'query': {
+        'bool': {
+        'filter': {
+            'term': {
+                'grad_year': 2022
+                
+            }
+        }
+        }
+    },
+    'sort': {
+        'last_name': {
+            'order': 'asc'
+        }
+    }       
+}
+
+response = client.search(index: 'students', from: 0, size: 10, body: query)
+```
+
+## Multi search
+
+You can bulk several queries together and do a multi search using the `msearch` method. The following snippet searches for students whose GPAs are outside the 3.1&ndash;3.9 range:
```

Comment:
Change "do" to "perform" or "run". Same comment re: "multi-search". In the second sentence, is "snippet" the best word?